Add BasePropOptions 'unique' and 'index'

simonB12 opened the pull request:
Closes #11. Feedback appreciated.
BR

szokodiakos commented:
@simonB12 Thank you for your pull request. One notice is that this project uses spaced object style: `{ something }` instead of `{something}`. Can you please correct this small issue? Other than that nicely done 👍 .

simonB12 commented:
Thanks for the comment. Changed space notation.

In the meantime the unit test went broken because of new cases (Jane Doe). added from another pull request: fixed too :-)

szokodiakos commented:
Thanks for the PR fixes, looks good!
